src/os/bluestore/BlueStore.cc

```diff
@@ -15431,36 +15410,39 @@ void BlueStore::_do_write_small(
 	  }
           logger->inc(l_bluestore_write_small_pre_read);
 
-	  _buffer_cache_write(txc, b, b_off, bl,
-			      wctx->buffered ? 0 : Buffer::FLAG_NOCACHE);
+          bufferlist without_padding;
+          without_padding.substr_of(bl, head_pad, bl.length() - head_pad);
```

Comment:
True, I don't seem to understand the existence of padding here. I understand head_read and tail_read as a way of saying "read remainder of block so we don't overwrite with garbage" and I imagine padding is "put zeros where no data should exists". Nevertheless, I moved buffer_cache_write to the start of the `if block` so that we lose complexity of offset calculations`.